java.lang.NoClassDefFoundError caused by dependency conflict issue due to multiple versions of com.google.guava:guava:jar

Hi, there are multiple versions of **com.google.guava:guava:jar** in **com.hotels.styx:styx-api-testsupport:jar:0.7.7**. Due to Maven version management, only com.google.guava:guava:jar:16.0.1 can be loaded and the other versions will be shadowed during the packaging process.

However, method **<com.google.common.base.MoreObjects: java.lang.Object firstNonNull(java.lang.Object,java.lang.Object)>** only defined in the shadowed version com.google.guava:guava:jar:18.0. As a result, an exception will be thrown with the following stack trace when your project referencing the missing method.

### Stack trace
Exception in thread "main" java.lang.NoClassDefFoundError: com/google/common/base/MoreObjects
	at com.github.tomakehurst.wiremock.common.ClasspathFileSource.<init>(ClasspathFileSource.java:53)
	at com.github.tomakehurst.wiremock.core.WireMockConfiguration.usingFilesUnderClasspath(WireMockConfiguration.java:153)
	at Test1.main(Test1.java:6)
Caused by: java.lang.ClassNotFoundException: com.google.common.base.MoreObjects
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 3 more

### Dependency tree 
com.hotels.styx:styx-api-testsupport:jar:0.7.7
+- com.hotels.styx:styx-api:jar:0.7.7:compile
|  +- io.netty:netty-all:jar:4.1.15.Final:compile
|  +- io.reactivex:rxjava:jar:1.1.6:compile
|  +- org.hdrhistogram:HdrHistogram:jar:2.1.4:compile
|  +- io.dropwizard.metrics:metrics-core:jar:3.2.6:compile
|  |  \- (org.slf4j:slf4j-api:jar:1.6.1:compile - version managed from 1.7.6; omitted for duplicate)
|  +- io.dropwizard.metrics:metrics-healthchecks:jar:3.2.6:compile
|  |  \- (org.slf4j:slf4j-api:jar:1.6.1:compile - version managed from 1.7.22; omitted for duplicate)
|  +- **com.google.guava:guava:jar:16.0.1:compile**
|  +- com.fasterxml.jackson.core:jackson-databind:jar:2.7.5:compile
|  |  +- (com.fasterxml.jackson.core:jackson-annotations:jar:2.7.5:compile - version managed from 2.4.2; omitted for duplicate)
|  |  \- (com.fasterxml.jackson.core:jackson-core:jar:2.7.5:compile - version managed from 2.4.2; omitted for duplicate)
|  +- com.fasterxml.uuid:java-uuid-generator:jar:3.1.5:compile
|  \- org.hamcrest:hamcrest-all:jar:1.3:compile
\- com.github.tomakehurst:wiremock:jar:1.54:compile
   +- org.apache.httpcomponents:httpclient:jar:4.3.5:compile
   |  +- org.apache.httpcomponents:httpcore:jar:4.3.2:compile
   |  +- commons-logging:commons-logging:jar:1.1.3:compile
   |  \- commons-codec:commons-codec:jar:1.6:compile
   +- org.slf4j:slf4j-api:jar:1.6.1:compile
   +- xmlunit:xmlunit:jar:1.5:compile
   +- org.mortbay.jetty:jetty:jar:6.1.26:compile
   |  +- org.mortbay.jetty:jetty-util:jar:6.1.26:compile
   |  \- org.mortbay.jetty:servlet-api:jar:2.5-20081211:compile
   +- org.skyscreamer:jsonassert:jar:1.2.3:compile
   |  \- org.json:json:jar:20090211:compile
   +- com.jayway.jsonpath:json-path:jar:0.8.1:compile
   |  +- net.minidev:json-smart:jar:1.1.1:compile
   |  \- commons-lang:commons-lang:jar:2.6:compile
   +- com.fasterxml.jackson.core:jackson-annotations:jar:2.7.5:compile
   +- com.fasterxml.jackson.core:jackson-core:jar:2.7.5:compile
   +- **(com.google.guava:guava:jar:16.0.1:compile - version managed from 18.0; omitted for duplicate)**
   +- net.sf.jopt-simple:jopt-simple:jar:4.7:compile
   \- (com.fasterxml.jackson.core:jackson-databind:jar:2.7.5:compile - version managed from 2.4.2; omitted for duplicate)

### Solution
Upgrade guava to 18.0 in parent-pom.

Thanks a lot.

Regards,
Leo

--- a/pom.xml
+++ b/pom.xml
@@ -92,7 +92,7 @@
     <netty-tcnative.classifier>linux-x86_64</netty-tcnative.classifier>
 
     <!-- General Versions -->
-    <guava.version>16.0.1</guava.version>
+    <guava.version>18.0</guava.version>
     <guice.version>3.0</guice.version>
     <jackson.version>2.7.5</jackson.version>
     <jackson.dataformat.version>2.5.1</jackson.dataformat.version>

